src/main/scala/org/codeoverflow/chatoverflow/requirement/service/tipeeestream/TipeeeStreamConnector.scala

```diff
@@ -1,19 +1,104 @@
 package org.codeoverflow.chatoverflow.requirement.service.tipeeestream
 
+import java.util.function.Consumer
+
+import io.socket.client.{IO, Socket}
+import org.codeoverflow.chatoverflow.WithLogger
+import org.codeoverflow.chatoverflow.api.io.dto.event.tipeeestream.{TipeeeStreamDonation, TipeeeStreamEvent, TipeeeStreamFollow, TipeeeStreamSubscription}
 import org.codeoverflow.chatoverflow.connector.Connector
+import org.json.{JSONException, JSONObject}
 
-class TipeeeStreamConnector(override val sourceIdentifier: String) extends Connector(sourceIdentifier) {
+import scala.collection.mutable.ListBuffer
 
-  override protected var requiredCredentialKeys: List[String] = List()
+/**
+  * The tipeeestream connector connects to the socket.io service to work with incoming events.
+  *
+  * @param sourceIdentifier the name of the tipeeestream account
+  */
+class TipeeeStreamConnector(override val sourceIdentifier: String) extends Connector(sourceIdentifier) with WithLogger {
+  private val eventHandler = ListBuffer[Consumer[TipeeeStreamEvent]]()
+  private val apiKey = "apiKey"
+  private val username = "username"
+  override protected var requiredCredentialKeys: List[String] = List(apiKey, username)
   override protected var optionalCredentialKeys: List[String] = List()
+  private var socket: Socket = _
```

Comment:
I prefer to have an empty optional here for two reasons: 
1. Eliminating null values from the code (this prevents a lot of trouble in general, altought in this special case it isn't a problem)
2. Consistency with other services (just have a look how it is done in [DiscordConnector](https://github.com/codeoverflow-org/chatoverflow/blob/master/src/main/scala/org/codeoverflow/chatoverflow/requirement/service/discord/DiscordChatConnector.scala) )